packages/nimble-components/src/rich-text/models/markdown-serializer.ts

```diff
@@ -9,12 +9,20 @@ import type { Node } from 'prosemirror-model';
  * Provides markdown serializer for rich text components
  */
 export class RichTextMarkdownSerializer {
+    private static mentionedUsers: string[];
+
     private static readonly markdownSerializer = this.initializeMarkdownSerializerForTipTap();
 
     public static serializeDOMToMarkdown(doc: Node): string {
+        RichTextMarkdownSerializer.mentionedUsers = [];
         return this.markdownSerializer.serialize(doc);
     }
 
+    public static getMentionedUser(doc: Node): string[] {
+        RichTextMarkdownSerializer.serializeDOMToMarkdown(doc);
```

Comment:
For viewer we will be utilizing getMentionedHrefs from parser as it has static markdown, and it won't get updated once parsed. So, the href will be taken from the parser itself.  But for editor, the state will be dynamically loaded, Consider this case where markdown we set can have one user and other users can be dynamically added to editor through UI, to get all the existing mentionedHrefs from the state, we need to get the list from the serializer. So whenever clients call `getMentionedhrefs()` method from configuration, In editor it will again serialize the current editor document and send us the mentioned hrefs. 

We can't move `mentionedUsers = []` into `getMentionedUser` method as we need to empty the array on each serialization, if not the array get populated on each `getMarkdown()` method call. Until the client didn't call `getMentionedUser` for the array to be updated. So to empty the array, we just kept under `serializeDOMToMarkdown()`.

Reply:
> if not the array get populated on each `getMarkdown()` method call.

Items are only added to that array if they do not already exist in it. Plus, it will only grow excessively if old mentions keep being deleted (but the array would still contain them), while new, different mentions keep being added. So there isn't really a memory/performance issue with moving `mentionedUsers = []`. And since nothing other than `getMentionedUsers` actually cares about the array being up-to-date, there's no functional issue either. So for the sake of clarity, I recommend puting both the reset and the return of the array in `getMentionedUsers`.

Reply:
> In the parser, calling getMentionedUsers just returns whatever users were collected during the last parse. Should we follow that same pattern here?

FYI I think the parser implementation is wrong so we shouldn't ncessarily copy it here. See [my comment on the parser](https://github.com/ni/nimble/pull/1683#discussion_r1406924013).

Reply:
> sake of clarity, I recommend puting both the reset and the return of the array in getMentionedUsers
 
 Moved it to `getMentionedUsers()`.

Reply:
As we discussed on the call this is probably an acceptable approach for now but we should verify how often `getMentionedUsers()` will be called and how long it takes to return. And we should file a tech debt item to see if we can get the same info from tiptap's tree without reserializing to markdown.

Reply:
Created tech debt https://github.com/ni/nimble/issues/1693 to track this approach.